state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -692,36 +738,19 @@ pub mod pallet {
 
 		/// Called when funds have been deposited into the given address.
 		///
-		/// Requires `EnsureWitnessed` origin.
+		/// Requires `EnsurePrewitnessed` or `EnsureWitnessed` origin.
 		#[pallet::call_index(2)]
 		#[pallet::weight(T::WeightInfo::process_single_deposit().saturating_mul(deposit_witnesses.len() as u64))]
```

Comment:
We should be able to do benchmark each function individually (`add_boostable_deposits` and `process_deposit_witnesses` ) and then add the condition of the origin inside this weight. e.g. you can do something like:
```rust
#[pallet::weight(if T::EnsurePrewitnessed::ensure_origin(origin.clone()).is_ok() {
			T::WeightInfo::process_boostable_deposits(deposit_witnesses.len() as u32)
		} else {
			T::WeightInfo::process_deposits_witnesses(deposit_witnesses.len() as u32)
		})]
```

Reply:
(could still use the `.saturating_mul` method too - main point is just that you can use the condition and benchmark each of the functions separately)

Reply:
RIP, seems like a limitation of substrate, will create an issue

Reply:
Think best we can do then is just use the more expensive direction here

Reply:
Issue already exists... and is 3.5 years old :/ https://github.com/paritytech/polkadot-sdk/issues/355

Reply:
Dang, I guess that the `process_deposit_witnesses` is the more expensive one, so ill just leave it as that for now.